### PR TITLE
fix: ledger catalyst voting

### PR DESCRIPTION
**Pull request by stackchain**
Resolves YOMO-564

Please note this needs to be tested using:

- Cardano APP v6 on Ledger
- Ledger needs to have experimental / developer features enabled

### CODE REVIEW

**Review by banklesss — APPROVED** [review]

File: apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts [hunk 1]
```diff
@@ -702,6 +702,12 @@ export const makeShelleyWallet = (constants: typeof MAINNET | typeof TESTNET) =>
       })
     }
 
+    async getFirstPaymentAddress() {
+      const externalAddress = this.externalAddresses[0]
+      const addr = await Cardano.Wasm.Address.fromBech32(externalAddress)
+      return Cardano.Wasm.BaseAddress.fromAddress(addr)
+    }
```
Comment (michalszorad) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts:709]: @stackchain please let me know if this is what you meant. Thanks!

Comment (stackchain) [reply apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts:709]: yap that is the idea, than later you can`this.getAddressing` and return the addressing obj to use it inside the voting.

Comment (michalszorad) [reply apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts:709]: So instead of `SHELLEY_VOTING_KEY_PATH` it would be `addressing.path`?

Comment (stackchain) [reply apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts:709]: yap, can you write some tests too?

File: apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/tests/mainnet/ShelleyWallet.test.ts [hunk 2]
```diff
@@ -332,6 +335,62 @@ describe('ShelleyWallet', () => {
...
+    const isReadOnly = false
+
+    const wallet: YoroiWallet & Record<string, any> = await ShelleyWalletMainnet.createBip44({
+      id: walletMeta.id,
+      accountPubKeyHex,
+      storage: storage.join(`${walletMeta.id}/`),
+      hwDeviceInfo,
+      isReadOnly,
+    })
+    wallet._utxos = [
```
Comment (michalszorad) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/tests/mainnet/ShelleyWallet.test.ts:361]: A hacky way here to modify utxos

Comment (michalszorad) [reply apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/tests/mainnet/ShelleyWallet.test.ts:361]: @stackchain is there any better way?

Comment (stackchain) [reply apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/tests/mainnet/ShelleyWallet.test.ts:361]: I believe we can use the current mocks, or also inject the `UtxoManager`, for now it's ok, we can improve the tests later.

**Pull request merged**